server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -561,6 +560,39 @@ public void test_can_generate_return_values() throws Exception {
         assertThat(returnValues).containsExactly("dummy-id-1", 10, 20);
     }
 
+    @Test
+    public void test_return_values_referring_to_non_deterministic_columns_use_persisted_value() throws Exception {
+        SQLExecutor e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (x int, y int default round((random() + 1) * 100), z int as round((random() + 1) * 100))")
+            .build();
+
+        DocTableInfo table = e.resolveTableInfo("tbl");
```

Comment:
I confirmed that reverting fix make this test fail. 

Should I remove integration test?

Reply:
Although adding itests is not preferrable, If we don't have another one to test this behaviour I'd prefer to have it as well on top of the unit test of Indexer.